multiversx_sdk_core/transaction_factories/delegation_factory.py

```diff
@@ -82,398 +66,266 @@ def add_nodes(self,
             receiver=delegation_contract,
             data_parts=parts,
-            execution_gas_limit=self.config.gas_limit_delegation_operations + self.config.additional_gas_for_delegation_operations,
-            gas_limit_hint=gas_limit,
-            gas_price=gas_price,
-            nonce=transaction_nonce,
-            value=value,
-            guardian=guardian
+            execution_gas_limit=self.config.gas_limit_delegation_operations + self.config.additional_gas_for_delegation_operations
         )
 
         return transaction
 
-    def set_metadata(self,
-                     sender: IAddress,
-                     delegation_contract: IAddress,
-                     name: str,
-                     website: str,
-                     identifier: str,
-                     value: ITransactionValue,
-                     guardian: Optional[IAddress] = None,
-                     transaction_nonce: Optional[INonce] = None,
-                     gas_price: Optional[IGasPrice] = None,
-                     gas_limit: Optional[IGasLimit] = None) -> Transaction:
+    def create_transaction_intent_for_setting_metadata(self,
+                                                       sender: IAddress,
+                                                       delegation_contract: IAddress,
+                                                       name: str,
+                                                       website: str,
+                                                       identifier: str) -> TransactionIntent:
         parts = [
             "setMetaData",
             arg_to_string(name),
             arg_to_string(website),
             arg_to_string(identifier)
         ]
 
-        transaction = self.create_transaction(
+        transaction = self._create_transaction_intent(
             sender=sender,
             receiver=delegation_contract,
             data_parts=parts,
-            execution_gas_limit=self.config.gas_limit_delegation_operations + self.config.additional_gas_for_delegation_operations,
-            gas_limit_hint=gas_limit,
-            gas_price=gas_price,
-            nonce=transaction_nonce,
-            value=value,
-            guardian=guardian
+            execution_gas_limit=self.config.gas_limit_delegation_operations + self.config.additional_gas_for_delegation_operations
         )
 
         return transaction
 
-    def _compute_gas_limit(self, payload: ITransactionPayload, execution_gas: IGasLimit, has_guardian: bool) -> IGasLimit:
+    def _compute_gas_limit(self, payload: ITransactionPayload, execution_gas: int) -> int:
         data_movement_gas = self.config.min_gas_limit + self.config.gas_limit_per_byte * payload.length()
         gas = data_movement_gas + execution_gas
 
-        if has_guardian:
-            gas += self.config.extra_gas_limit_for_guarded_transactions
+        # add by default extra gas for guarded transactions
+        gas += DEFAULT_EXTRA_GAS_LIMIT_FOR_GUARDED_TRANSACTION
```

Comment:
Maybe we should shift the responsibility of increasing the gas limit to the caller, who gets the created transaction intent and converts it into a transaction, then triggers the signing flow. That is, I think we can remove this logic. And perhaps also specify in the specs repo that callers / clients are responsible with patching the gas limit regarding guarded transactions?

Reply:
Indeed! Removed the extra gas for guarded transactions. Will also mention in the specs.